kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+                
+            }                            
+        }
+
+        return true;
+    }
+    
+    /**
+     * Gets the default Chart Code, Account from the custom Procurement Cardholder table.
+     * 
+     */
+    private ProcurementCardDefault getProcurementCardDefault(String creditCardNumber) {
+                
+        Map<String, String> pkMap = new HashMap<String, String>();
+        pkMap.put("creditCardNumber", creditCardNumber);
+        ProcurementCardDefault procurementCardDefault = (ProcurementCardDefault) businessObjectService.findByPrimaryKey(ProcurementCardDefault.class, pkMap);
+                
+        return procurementCardDefault;
+    }
+    
+    private void requeueDocument(ProcurementCardDocument pcardDocument, String pcardDocumentId, String node, String prevNode) {
+        Long documentId = Long.valueOf(pcardDocumentId);
+        WorkflowDocumentActionsService documentActions = KewApiServiceLocator.getWorkflowDocumentActionsService();
+        // Updated System User ID, should change this to use the parameter
+        DocumentActionParameters actionParameters = DocumentActionParameters.create(Long.toString(documentId), "T000000000000005493", node+" Batch Reroute to Reconciler");
+        ReturnPoint returnPoint = ReturnPoint.create(prevNode);
+        documentActions.superUserReturnToPreviousNode(actionParameters, false, returnPoint);
+        // Use requeuer service to put the document back into action list of reconciler
+        DocumentRefreshQueue docRequeue = KewApiServiceLocator.getDocumentRequeuerService("KFS",Long.toString(documentId), 0x0);
+        docRequeue.refreshDocument(Long.toString(documentId));    
+       
+        if ( LOG.isInfoEnabled() ) {
+            LOG.info("Rerouting PCDO document # " + pcardDocumentId + ".");
+        }
+    }
+    
+    /*
+     * as the edu.arizona ProcurementCardDocument implentation has additional fields, we must duplicate this method from the extended class and use the edu.arizona.ProcurementCardDocument implementation(non-Javadoc)
+     * @see org.kuali.kfs.fp.batch.service.impl.ProcurementCardCreateDocumentServiceImpl#createProcurementCardDocuments()
+     */
+    @Override
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Transactional
+    public boolean createProcurementCardDocuments() {
+        List documents = new ArrayList();
+        List cardTransactions = retrieveTransactions();
+
+        // iterate through card transaction list and create documents
+        for (Iterator iter = cardTransactions.iterator(); iter.hasNext();) {
+            documents.add(this.createProcurementCardDocument((List) iter.next()));
+        }
+
+        // now store all the documents
+        for (Iterator iter = documents.iterator(); iter.hasNext();) {
+            ProcurementCardDocument pcardDocument = (ProcurementCardDocument) iter.next();
+            try {
+                documentService.saveDocument(pcardDocument, DocumentSystemSaveEvent.class);
```

Comment:
I'm not sold on this... if a getter should be used as though the documentService was being fetched from an outside class, what's the point of extending the bases class and having the visibility of documentService opened up to 'protected'?